sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/application-offering-change-request.students.controller.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  Body,
+  Controller,
+  Get,
+  NotFoundException,
+  Param,
+  ParseIntPipe,
+  Patch,
+  UnauthorizedException,
+} from "@nestjs/common";
+import {
+  ApiNotFoundResponse,
+  ApiTags,
+  ApiUnauthorizedResponse,
+} from "@nestjs/swagger";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { AllowAuthorizedParty, UserToken } from "../../auth/decorators";
+import { ApiProcessError, ClientTypeBaseRoute } from "../../types";
+import BaseController from "../BaseController";
+import {
+  ApplicationOfferingChangesAPIOutDTO,
+  UpdateApplicationOfferingChangeRequestAPIInDTO,
+} from "./models/application-offering-change-request.dto";
+import { ApplicationOfferingChangeRequestStatus } from "@sims/sims-db";
+import { StudentUserToken } from "../../auth";
+import { STUDENT_UNAUTHORIZED_FOR_APPLICATION_OFFERING_CHANGE_REQUEST } from "../../constants";
+import { ApplicationOfferingChangeRequestControllerService } from "./application-offering-change-request.controller.service";
+import { ApplicationOfferingChangeRequestService } from "../../services";
+
+/**
+ * Application offering change request controller for students client.
+ */
+@AllowAuthorizedParty(AuthorizedParties.student)
+@Controller(
+  "application-offering-change-request/:applicationOfferingChangeRequestId",
+)
+@ApiTags(`${ClientTypeBaseRoute.Student}-application-offering-change-request`)
+export class ApplicationOfferingChangeRequestStudentsController extends BaseController {
+  constructor(
+    private readonly applicationOfferingChangeRequestService: ApplicationOfferingChangeRequestService,
+    private readonly applicationOfferingChangeRequestControllerService: ApplicationOfferingChangeRequestControllerService,
+  ) {
+    super();
+  }
+
+  /**
+   * Gets the Application Offering Change Request details.
+   * @param applicationOfferingChangeRequestId the Application Offering Change Request id.
+   * @param studentUserToken student user token to authorize the user.
+   * @returns Application Offering Change Request details.
+   */
+  @Get()
```

Comment:
I would say you can use the param `:applicationOfferingChangeRequestId` in Get() e.g. Get(':applicationOfferingChangeRequestId') for better readability as it is used in line57 instead of controller root.

Also if we are going to have any API without ('::applicationOfferingChangeRequestId') in url, then we have to do a refactoring to all existing methods.